src/analysis/inner_product_space/spectrum.lean

```diff
@@ -242,5 +242,27 @@ end
 
 end version2
 
+section positive
+variables {n : ℕ} (hn : finite_dimensional.finrank 𝕜 E = n)
+
+lemma pos_nonneg_eigenvalues (hpos : ∀ (x : E), 0 ≤ is_R_or_C.re ⟪T x, x⟫) :
+  ∀ (i : (fin n)), 0 ≤ hT.eigenvalues hn i :=
+begin
+  intro i,
+  have : hT.eigenvalues hn i =
+    is_R_or_C.re ⟪ T (hT.eigenvector_basis hn i), hT.eigenvector_basis hn i ⟫ :=
+  begin
+    simp only [inner_smul_left, inner_product_space.is_self_adjoint.apply_eigenvector_basis,
+      is_R_or_C.conj_of_real, is_R_or_C.mul_re, is_R_or_C.of_real_re, sub_zero, mul_zero,
+      inner_self_nonneg_im, is_R_or_C.of_real_im, inner_self_eq_norm_sq_to_K],
+    rw (hT.eigenvector_basis_orthonormal hn).1,
+    norm_num,
+  end,
+  rw this,
+  exact (hpos (hT.eigenvector_basis hn i)),
```

Comment:
Two comments: when using `have` and going into tactic mode, just focus with braces `{ }` instead of `:= begin ... end`. Also, I just golfed the proof a minor amount. If you want, you can squeeze the simp, but it's not necessary.
```suggestion
  have : is_R_or_C.re ⟪ T (hT.eigenvector_basis hn i), hT.eigenvector_basis hn i ⟫ 
    = hT.eigenvalues hn i,
  { simp [inner_smul_left, inner_self_eq_norm_sq_to_K, (hT.eigenvector_basis_orthonormal hn).1] },
  exact this ▸ hpos (hT.eigenvector_basis hn i),
```

Reply:
I see, I've been using begin/end for `have`, but I'm happy to use braces instead.  I'll also golf a bit more myself going forward.